aodntools/timeseries_products/Documentation/aggregated_timeseries.md

```diff
@@ -0,0 +1,175 @@
+- All files to be aggregated are from the same site, and have the same site_code attribute;
+- Variables to be aggregated have TIME as their only dimension (or if LATITUDE and LONGITUDE are included as dimensions, they have size 1);
+- Variables to be aggregated do not represent current velocity;
+- The in-water data are bounded by the global attributes time_deployment_start and time_deployment_end.
+
+This project will not make any corrections to or implement work-arounds for input files that do not meet these requirements. A report on files that cannot be aggregated will be produced. If any files are re-processed by the moorings facility to meet the requirements, these will be incorporated into future versions of the product.
+
+While the code will be more generally applicable, the set of product files created as part of this project will be limited to the following variables (standard/long name in brackets):
+
+- TEMP (sea_water_temperature);
+- PSAL (sea_water_salinity);
+- CPHL, CHLF, CHLU (mass_concentration_of_inferred_chlorophyll_from_relative_fluorescence_units_in_sea_water) - considered as separate variables for now;
+- TURB (sea_water_turbidity);
+- DOX1, DOX1_2, DOX1_3 (mole_concentration_of_dissolved_molecular_oxygen_in_sea_water);
+- DOX2 (moles_of_oxygen_per_unit_mass_in_sea_water);
+- DOXS (fractional_saturation_of_oxygen_in_sea_water).
+- PAR (downwelling_photosynthetic_photon_flux)
+
+## Method
+
+Generating function: 
+
+```
+usage: aggregated_timeseries.py [-h] -var VARNAME -site SITE_CODE -files
+                                FILENAMES [-path OUTPUT_PATH]
+
+Concatenate ONE variable from ALL instruments from ALL deployments from ONE
+site
+
+optional arguments:
+  -h, --help         show this help message and exit
+  -var VARNAME       name of the variable to concatenate. Like TEMP, PSAL
+  -site SITE_CODE    site code, like NRMMAI
+  -files FILENAMES   name of the file that contains the source URLs
+  -path OUTPUT_PATH  path where the result file will be written. Default ./
+
+```
+
+
+
+
+### Input file selection
+
+The input files are accessed using the OPeNDAP protocol via the AODN THREDDS server.  A list of OPeNDAP URLs can be produced by the utility function get_moorings_url(), which queries an index of all published moorings files to obtain the files relevant for a given site and variable of interest (VoI). The list can also be filtered by feature type (e.g time-series or profile) temporal range, file version (FV01), processing mode (real-time/delayed) or a regular expression matched against the file name.
+
+### Input file validation
+
+Before proceeding to the aggregation, each input file will be checked to ensure it meets the requirements (as specified above under Inputs). This will prevent errors during the aggregation process. If any files fail to meet the requirements, a list of all non-compliant files is output, and the code stops. An option can be provided to continue processing only the compliant files.
```

Comment:
Again, this is out of date. We later changed this to:
> ... Any input files that fail to meet the requirements will be excluded from the aggregation, and their URL listed in a global attribute rejected_files.

Reply:
fixed